sktime/forecasting/compose/_pipeline.py

```diff
@@ -1315,7 +1327,11 @@ def _fit(self, y, X, fh):
             self.forecaster_X_.fit(y=self._get_Xcols(X), fh=fh_X)
 
         self.forecaster_y_ = self.forecaster_y.clone()
-        self.forecaster_y_.fit(y=y, X=X, fh=fh)
+        if self.fit_behaviour == "use_gt":
+            self.forecaster_y_.fit(y=y, X=X, fh=fh)
+        elif self.fit_behaviour == "use_forecast":
+            x_insample = self.forecaster_X_.predict(fh=X.index, X=X)
```

Comment:
Not all of the models support in-sample prediction, there's a boolean tag `capability:insample` for this.

Reply:
Thanks. I added a check.